docs/contributors/folder-structure.md

```diff
@@ -0,0 +1,192 @@
+    │
+    ├── phpunit.xml.dist
+    │   Config file for PHPUnit.
+    │
+    ├── postcss.config.js
+    │   Config file for PostCSS.
+    │
+    ├── renovate.json
+    │   Config file for Renovate.
+    │
+    ├── docker-compose.yml
+    │   Docker config files for the development and testing environment.
+    │
+    ├── .env
+    ├── .wp-env.json
+    │   Config files for the development and testing environment.
+    │   Includes WordPress, the WooCommerce plugin and the Storefront theme.
+    │
+    ├── composer.lock
+    ├── composer.json
+    │   Handling of PHP dependencies. Essentially used for development tools.
+    │   The production code don't use external PHP dependencies.
+    │
+    ├── package-lock.json
+    ├── package.json
+    │	Handling of JavaScript dependencies. Both for development tools and
+    │   production dependencies. The package.json also serves to define common
+    |   tasks and scripts used for day to day development.
+    │
+    ├── readme.txt
+    │   Readme of the WooCommerce Blocks plugin hosted on the WordPress
+    │   plugin repository.
+    │
+    ├── woocommerce-gutenberg-products-block.php
+    │   Entry point of the WooCommerce Blocks plugin.
+    │
+    ├── .github/*
+    │   Config of the different Github features (issues and PR templates, CI, 
+    │   owners).
+    │
+    ├── .sources/*
+    │   Sketch files of the WooCommerce Blocks plugin.
+    │
+    ├── .wordpress-org/*
+    │   Assets of the WooCommerce Blocks plugin hosted on the WordPress plugin
+    │   repository.
+    │
+    ├── assets/css/*
+    │   The SCSS files of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/*
+    │   The React components of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/atomic
+    │   The atomic components such as product title, product rating, product
+    │   image, etc. These atomic components are used by various blocks.
+    │
+    ├── assets/js/base
+    │   Functionality to ensure that the frontend does not get too bloated.
+    │
+    ├── assets/js/blocks-registry
+    │   Files that allows developers to connect their extensions to this plugin.
+    │
+    ├── assets/js/blocks
+    │   The main blocks of the WooCommerce Blocks plugin such as the Active 
+    │   Filters Block.
+    │
+    ├── assets/js/data
+    │   Functionality to store data using Redux and wp.data.
+    │
+    ├── assets/js/editor-components
+    │   Editor components such as the block-title component.
+    │
+    ├── assets/js/extensions
+    │   TypeScript files to allow Google Analytics tracking of specific events
+    │   such as active payment method and removing cart items.
+    │
+    ├── assets/js/filters
+    │   Filter such as excluding the checkout draft from Google Analytics.
+    │
+    ├── assets/js/hocs
+    │   The Higher Order Components of the WooCommerce Blocks plugin.
+    │
+    ├── assets/js/icons
+    │   The SVG icons of the WooCommerce Blocks plugin.
```

Comment:
```suggestion
    │   The WooCommerce Blocks Icon library.
```

Reply:
Thanks for this suggestion, @mikejolley!